docs/DeveloperGuide.adoc

```diff
@@ -1215,74 +1235,119 @@ Use case ends.
 Use case ends.
 
 [discrete]
-=== Use case: Apply leave
+=== Use case: Apply leave application
 *MSS*
 
-.	User enters command to apply off day
-.	AddressBook list the remaining off days left and prompts the date the user wants to apply on
-.	User enters date
-.	AddressBook add the request to the database for approval
+.	User enters `leave` command with the desired leave application date.
+.	BankAB adds the leave application to the database for approval.
 +
 Use case ends.
 
 *Extensions*
 
-* 2a. No remaining days left
+* 2a. User not logged in
+** 2a1. BankAB shows an error message.
++
+User case ends.
+
+* 2b. The given date is an invalid calendar date.
+** 2b1. BankAB shows an error message.
 +
 Use case ends.
 
-* 4a. The given date is not available for application
-*  4a1. AddressBook shows an error message.
+* 2c. The given date has passed.
+** 2c1. BankAB shows an error message.
 +
-Use case resumes to step 2.
+Use case ends.
+
+* 2d. Today's date is given.
+** 2d1. BankAB shows an error message.
++
+Use case ends.
+
+* 2e. The given date is an existing leave application.
+** 2e1. BankAB shows an error message.
++
+Use case ends.
 
 [discrete]
-=== Use case: List leave
+=== Use case: List leave application
+
+*MSS*
+
+.	User enters `listleave` command.
+.	BankAB list all leave applications.
++
+Use case ends.
 
 [discrete]
-=== Use case: Filter leave
+=== Use case: Filter leave application
+
+*MSS*
+
+.	User enters `filterleave` command with the desired NRIC he wants to filter.
+.	BankAB list all leave applications under that NRIC.
++
+Use case ends.
```

Comment:
For filter leave can add Extensions such that when the NRIC typed does not exist, it shows an empty list.

Reply:
Noted.